Adopt in-process AVAssetWriter on platforms supporting it

jyavenard opened the pull request:
#### c182d09d383ec8ba78e101e927eee370a5bf96a0
<pre>
Adopt in-process AVAssetWriter on platforms supporting it
<a href="https://bugs.webkit.org/show_bug.cgi?id=295377">https://bugs.webkit.org/show_bug.cgi?id=295377</a>
<a href="https://rdar.apple.com/154912553">rdar://154912553</a>

Reviewed by Jer Noble.

We disable the use of the GPU process&apos;s MediaPlayerPrivateWriter proxy
on platforms not requiring it.

Covered by existing tests.

* LayoutTests/ipc/invalid-mediarecorder-identifier.html: The IPC MediaRecorder service is only enabled on real iOS 18/visionOS 2.0 devices
and not on simulators, so we can&apos;t test it in our existing infrastructure. We can remove it altogether.
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/MediaStrategy.cpp:
(WebCore::MediaStrategy::createMediaRecorderPrivateWriter const):
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateWriter.cpp:
(WebCore::MediaRecorderPrivateWriter::create):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::GPUConnectionToWebProcess::allowsExitUnderMemoryPressure const):
(WebKit::GPUConnectionToWebProcess::dispatchMessage):
(WebKit::GPUConnectionToWebProcess::dispatchSyncMessage):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.h:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.h:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.cpp:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:

Canonical link: <a href="https://commits.webkit.org/297009@main">https://commits.webkit.org/297009@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/27c13d762522112ebce855506ad47498cff8607c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/110230 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/29889 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/20321 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/116254 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/60478 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/112193 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/30567 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/38476 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/83809 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/113178 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/24385 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/99270 "Passed tests") | [❌ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/64256 "Found 1 new API test failure: /WPE/TestSSL:/webkit/WebKitWebView/ephemeral-tls-errors (failure)") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/23752 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/17405 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/60048 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/102721 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/93761 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/17464 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/119044 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/108784 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/37270 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/27636 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/92778 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/37642 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/95540 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/92604 "Found 2 new API test failures: /WebKitGTK/TestWebProcessExtensions:/webkit/WebKitWebView/web-process-crashed, /WebKitGTK/TestWebKitPolicyClient:/webkit/WebKitPolicyClient/new-window-policy (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23602 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/37602 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/15338 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/33159 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/37164 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/42635 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/133059 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/36826 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35970 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/40166 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/38535 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/7fd5cd0d1f9d848c9f65c8a76a2663cc39a5045d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47525

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/27c13d762522112ebce855506ad47498cff8607c. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/47525

jyavenard committed:
Adopt in-process AVAssetWriter on platforms supporting it
https://bugs.webkit.org/show_bug.cgi?id=295377
rdar://154912553

Reviewed by Jer Noble.

We disable the use of the GPU process's MediaPlayerPrivateWriter proxy
on platforms not requiring it.

Covered by existing tests.

* LayoutTests/ipc/invalid-mediarecorder-identifier.html: The IPC MediaRecorder service is only enabled on real iOS 18/visionOS 2.0 devices
and not on simulators, so we can't test it in our existing infrastructure. We can remove it altogether.
* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/platform/MediaStrategy.cpp:
(WebCore::MediaStrategy::createMediaRecorderPrivateWriter const):
* Source/WebCore/platform/MediaStrategy.h:
* Source/WebCore/platform/mediarecorder/MediaRecorderPrivateWriter.cpp:
(WebCore::MediaRecorderPrivateWriter::create):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.cpp:
(WebKit::GPUConnectionToWebProcess::allowsExitUnderMemoryPressure const):
(WebKit::GPUConnectionToWebProcess::dispatchMessage):
(WebKit::GPUConnectionToWebProcess::dispatchSyncMessage):
* Source/WebKit/GPUProcess/GPUConnectionToWebProcess.h:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.cpp:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.h:
* Source/WebKit/GPUProcess/media/RemoteMediaRecorderPrivateWriterManager.messages.in:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.cpp:
* Source/WebKit/WebProcess/GPU/media/RemoteMediaRecorderPrivateWriter.h:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.cpp:
* Source/WebKit/WebProcess/GPU/media/WebMediaStrategy.h:

Canonical link: https://commits.webkit.org/297009@main
